File: search-insert-position/search-insert-position.js
const binarySearch = function (arr, target, left, right) {
  let mid;
  while (left<=right) {
    mid = Math.floor((left+right) / 2);
    if (arr[mid]===target){
      return mid;
    } else if (target>arr[mid]){
      left = mid+1;
    } else {
      right = mid-1;
    }
  }
  if (target === 0)
    return 0;
  if (target <= arr[mid])
    return mid;
  return mid+1;
}

const searchInsert = function (nums, target) {
  return binarySearch(nums, target, 0, nums.length-1)
};

File: search-insert-position/search insert position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
    int l=0,r=nums.size()-1,m,i;
    while (l <= r) {
        m =l + (r - l)/ 2;
  
        // Check if x is present at mid
        if (nums[m] == target)
            return m;
  
        // If x greater, ignore left half
        if (nums[m] < target)
            l = m + 1;
  
        // If x is smaller, ignore right half
        else
            r = m - 1;
    }
    for(i=0;i<nums.size();i++)
    {
        if(nums[i]>target)
        {
            return i;
        }
    }
    return i;
        }
};